xds/internal/xdsclient/tests/lds_watchers_test.go

```diff
@@ -742,7 +742,9 @@ func (s) TestLDSWatch_ValidResponseCancelsExpiryTimerBehavior(t *testing.T) {
 // scenarios:
 //  1. Removing a resource should trigger the watch callback with a resource
 //     removed error. It should not trigger the watch callback for an unrelated
-//     resource.
+//     resource. If any new watcher try to register watcher for removed
```

Comment:
I know that we haven't always done a good job with keeping tests focussed on one thing. So some of our tests do multiple things. But it would be nice to have a separate test for this new scenario instead of adding more to an existing test (which looking at it now, is already doing a lot).

See: https://engdoc.corp.google.com/eng/doc/tott/episodes/726.md?cl=head

Reply:
Added a separate test